src/Chapter4.hs

```diff
@@ -489,6 +506,25 @@ Implement the 'Applicative' instance for our 'List' type.
   type.
 -}
 
+instance Applicative List where
+  pure :: a -> List a
+  pure x = Cons x Empty
+
+  (<*>) :: List (a -> b) -> List a -> List b
+  -- Original version
+  -- Empty <*> _ = Empty
+  -- (Cons f fs) <*> xs = (fmap f xs) `myAppend` (fs <*> xs)
+  fs <*> xs = myConcat $ fmap (\f -> fmap f xs) fs
```

Comment:
Your original version is already good enough :+1: 
But the new one is clever :brain: